fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,180 @@
+                metrics.append(metric)
+
+
+    if axs is None:
+        if subplot_shape is None:
+            subplot_shape = (1, len(metrics))
+        fig, axs = plt.subplots(*subplot_shape, squeeze=False, figsize=figsize)
+        axs = axs.flatten()
+
+    # plotting without error
+    df['_index'] = df.index
+    if error_bars is None and conf_intervals is None:
+        for metric, ax in zip(metrics, axs):
+            previous_xlabel = ax.get_xlabel()
+            ax = df.plot(x="_index", y=metric, kind=plot_type, ax=ax, colormap=colormap)
+            ax.set_xlabel(previous_xlabel)
+        del df['_index']
+        return axs
+
+    df_all_errors = pd.DataFrame([])
+    df_all_bounds = pd.DataFrame([])
+    # plotting with confidence intervals:
+    if conf_intervals is not None:
+        for metric, conf_interval in zip(metrics, conf_intervals):
+            df_all_errors[metric] = abs(df[metric] - df[conf_interval])
+
+            # assert bounds are valid
+            #assert (df_lower_error >= 0).all(), _NEG_ERROR_OR_FLIPPED_BOUNDS_ERROR
+
+            if plot_error_labels:
+                df_all_bounds[metric] = df[conf_interval]
+    # plotting with relative errors
+    elif error_bars is not None:
+        for metric, error_bar in zip(metrics, error_bars):
+            df_all_errors[metric] = df[error_bar]
+
+            # assert bounds are valid
+            #assert (df_error >= 0).all(), _NEG_ERROR_OR_FLIPPED_BOUNDS_ERROR
+
+            if plot_error_labels:
+                df_error = pd.DataFrame([])
+                df_error[['lower', 'upper']] = pd.DataFrame(df[error_bar].tolist(), index=df.index)
+                df_error['error'] = list(zip(df[metric] - df_error['lower'], df[metric] + df_error['upper']))
+                df_all_bounds[metric] = df_error['error']
+    
+    for metric, ax in zip(metrics, axs):
+        previous_xlabel = ax.get_xlabel()
+        ax = df.plot(x="_index", y=metric, kind=plot_type, yerr=df_all_errors, ax=ax, colormap=colormap)
+        ax.set_xlabel(previous_xlabel)
+    # TODO: Check assumption of plotting items in the vertical direction
+
+    if plot_error_labels:
+        for j, metric in enumerate(metrics):
+            y_min, y_max = axs[j].get_ylim()
+            y_range = y_max - y_min
+
+            # TODO: Figure out if works for other plotting modes (besides bar)
+            for i in range(len(axs[j].patches)):
+                axs[j].text(i, df_all_bounds[metric][i][0] - 0.05 * y_range, round(df_all_bounds[metric][i][0],
+                        text_precision_digits), fontsize=text_fontsize, color=text_color,
+                        ha=text_ha)
+                axs[j].text(i, df_all_bounds[metric][i][1] + 0.01 * y_range, round(df_all_bounds[metric][i][1],
+                        text_precision_digits), fontsize=text_fontsize, color=text_color,
+                        ha=text_ha)
+
+    if show_plot:
```

Comment:
Should the 'no error bars at all' case also hit this `if` statement?

Reply:
I'm actually not sure if I can stop `df.plot` from showing the plot. Even without the line plt.show(), it still shows.

Reply:
That may be due to other configuration, though. If I run the pandas plot function it doesn't show until I run `plt.show()` separately.